core/src/test/scala/org/apache/spark/scheduler/TaskSetBlacklistSuite.scala

```diff
@@ -110,7 +131,13 @@ class TaskSetBlacklistSuite extends SparkFunSuite {
       .set(config.MAX_TASK_ATTEMPTS_PER_NODE, 3)
       .set(config.MAX_FAILURES_PER_EXEC_STAGE, 2)
       .set(config.MAX_FAILED_EXEC_PER_NODE_STAGE, 3)
-    val taskSetBlacklist = new TaskSetBlacklist(conf, stageId = 0, new SystemClock())
+    val clock = new ManualClock
+
+    val attemptId = 0
+    val taskSetBlacklist = new TaskSetBlacklist(
+      listenerBusMock, conf, stageId = 0, stageAttemptId = attemptId, clock = clock)
+
+    clock.setTime(0)
```

Comment:
You should set the time to a new value before each call of taskSetBlacklist.updateBlacklistForFailedTask to see that the events on the listenerbus has the correct time.